File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.py
from collections import Counter 

def solution(participant, completion):
    answer = ''
    part_dict = Counter(participant)
    com_dict = Counter(completion)
    ans_dict = part_dict - com_dict
    
    for i in ans_dict :
        if ans_dict[i] == 1:
            answer = i
            break
            
    return answer

File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.java
import java.util.*;


class Solution {
    public String solution(String[] participant, String[] completion) {
        
        String answer = "";
        
        HashMap <String,Integer> count = new HashMap<>();
        
        for(String name : participant){
            count.put(name,count.getOrDefault(name,0)+1);
        }
        
        for(String name : completion){
            count.put(name,count.get(name)-1);
        }
        
        for(String key : count.keySet()){
            if(count.get(key)!=0){
                answer = key;
                break;
            }

        }
        
        
        return answer;
    }
}